tests/core/test_rule_groups.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+
+from smdebug_rulesconfig.debugger_rules._constants import (
+    SUPPORTED_DL_FRAMEWORKS,
+    SUPPORTED_FRAMEWORKS,
+)
+from smdebug_rulesconfig.debugger_rules._ruleGroups import (
+    DEEP_LEARNING_APPLICATION_RULES,
+    DEEP_LEARNING_RULES,
+    UNIVERSAL_RULES,
+    XGBOOST_RULES,
+)
+from smdebug_rulesconfig.debugger_rules._utils import _get_rule_list
+
+
+@pytest.mark.parametrize(
+    "rule_list, framework_list",
+    [
+        (UNIVERSAL_RULES, SUPPORTED_FRAMEWORKS),
+        (DEEP_LEARNING_RULES, SUPPORTED_DL_FRAMEWORKS),
+        (DEEP_LEARNING_APPLICATION_RULES, SUPPORTED_DL_FRAMEWORKS),
+        (XGBOOST_RULES, ["xgboost"]),
+    ],
+)
+def test_framework_rule_compatibility(rule_list, framework_list):
+    for rule in rule_list:
+        for framework in framework_list:
+            assert rule in _get_rule_list(framework), f"Rule: {rule} not supported by {framework}"
```

Comment:
To phrase this test in words, for each given rule list and framework list, we are checking that the rule list is a subset of every framework's rule list. i.e. the universal rules are included in every framework's rule list.

Then I have a minor suggestion for making the test more clear:
```
for framework in framework_list:
    assert set(rule_list).is_subset(set(_get_rule_list(framework)))
```

I'll leave the decision up to you. We should have a docstring for the test at least.